src/session.ts

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: Apache-2.0
+"use strict";
+
+import { ethers, Signer } from "ethers";
+
+import { TxReceipt, BalanceProof } from "#erdstall/api/responses";
+import { Transfer, Mint, ExitRequest, TradeOffer, Trade } from "#erdstall/api/transactions";
+import { EnclaveWriter } from "#erdstall/enclave";
+import { Address, LedgerWriter } from "#erdstall/ledger";
+import { Assets } from "#erdstall/ledger/assets";
+import { Uint256 } from "#erdstall/api/util";
+import { Stages } from "#erdstall/utils";
+import { ErdstallSession } from "./erdstall";
+import { Client } from "./client";
+
+export const ErrUnitialisedClient = new Error("client unitialised");
+
+export class Session extends Client implements ErdstallSession {
+	readonly address: Address;
+	private nonce: bigint;
+	private readonly enclaveWriter: EnclaveWriter;
+	private readonly signer: Signer;
+
+
+	static dial(
+		address: Address,
+		signer: Signer,
+		operatorAddress: URL
+	): Session {
+		return new Session(address, new Client(signer, operatorAddress));
```

Comment:
This would become sth like
```suggestion
  const encConn = new Enclave(new EnclaveWSProvider(operatorUrl));
		return new Session(address, signer, encConn);
```

Reply:
The whole constructor is no longer necessary.